lib/calculator.ex

```diff
@@ -7,6 +9,95 @@ defmodule Calculator do
   Calculator GenServer.
 
   ## Examples
-
+      iex> Calculator.start_link
+      iex> Calculator.call(:add, 1, 2)
+      3
   """
+
+  def start_link do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  @impl GenServer
+  def init(_args) do
+    {:ok, 0}
+  end
+
+  def call(operation, num_1, num_2) do
+    GenServer.call(__MODULE__, {operation, num_1, num_2})
+  end
```

Comment:
Showing guards here only (see the comment below.) `call/2` would be the same.

```suggestion
  def call(:divide, _num_1, 0) do
    :go_learn_arithmetics
  end
  def call(operation, num_1, num_2) when is_number(num_1) and is_number(num_2) do
    GenServer.call(__MODULE__, {operation, num_1, num_2})
  end
  def call(operation, _num_1, _num_2) do
        :go_learn_arithmetics
  end
```